smart-contracts/contracts/airdrop/src/admin.rs

```diff
@@ -0,0 +1,243 @@
+
+    // Check if the number of users and amounts provided match
+    if users.len() != amounts.len() {
+        return Err(AirdropErrors::Std(StdError::GenericErr {
+            msg: "Deposit amount weight for primitive is zero".to_string(),
+        }));
+    }
+
+    for number in 0..=users.len() {
+        // Validate the user's address
+        deps.api.addr_validate(&users[number].to_string())?;
+
+        // Validate that the amount is not negative
+        if amounts[number] < Uint128::zero() {
+            return Err(AirdropErrors::Std(StdError::GenericErr {
+                msg: "Amount at index :".to_string()
+                    + &*number.to_string()
+                    + &*"is negative".to_string(),
+            }));
+        }
+
+        // Load the user's current information from storage
+        let user_info = USER_INFO.load(deps.storage, users[number].clone())?;
+
+        // Check if the user has not claimed
+        if !user_info.get_claimed_flag() {
+            // Update all the users with the given info
+            USER_INFO.save(deps.storage, users[number].clone(), &user_info)?;
+        }
+    }
+
+    // Calculate the total claimable amount from USER_INFO
+    let total_in_user_info = get_total_in_user_info(deps.storage);
+
+    // Check if the total claimable amount exceeds the airdrop amount
+    if total_in_user_info > current_airdrop_config.airdrop_amount {
+        return Err(AirdropErrors::Std(StdError::GenericErr {
+            msg: "Total amount in the given user amounts".to_string()
+                + &*total_in_user_info.to_string()
+                + &*" is greater than ".to_string()
+                + &*current_airdrop_config.airdrop_amount.to_string(),
+        }));
+    }
+
+    // Return a default response if all checks pass
+    // TODO: Add events
+    Ok(Response::default())
+}
+
+pub fn execute_remove_users(deps: DepsMut, users: Vec<String>) -> Result<Response, AirdropErrors> {
+    // Load the current airdrop configuration from storage
+    let current_airdrop_config = AIRDROP_CONFIG.load(deps.storage)?;
+
+    // Check if the current airdrop window is not open (start_height or end_height not zero)
+    if current_airdrop_config.start_height != 0 || current_airdrop_config.end_height != 0 {
+        return Err(AirdropErrors::InvalidChangeUserInfo {});
+    }
+
+    // Iterate through the list of users to be removed
+    for user in users.iter() {
+        // Validate the user's address
+        deps.api.addr_validate(user)?;
+
+        // Load the user_info entry from storage
+        let user_info = USER_INFO.load(deps.storage, user.clone())?;
+
+        // Check if the claimed flag is false, indicating that the user has not claimed
+        if !user_info.get_claimed_flag() {
+            // Remove the user's entry from the USER_INFO map
+            USER_INFO.remove(deps.storage, user.clone());
+        }
+    }
+
+    // Return a default response if all checks pass
+    // TODO: Add events
```

Comment:
Todo should be added here, especially since `user_info.get_claimed_flag() == true` leads to no removal but an Ok response

Reply:
Oh okay. Added a separate array of only the removed users.